### PR TITLE
feat: add request_missing_tracks.py script

### PR DESCRIPTION
**Pull request by ca-scribner**
This script automates asking for the tracks referenced by a bundle that do not yet exist, outputting instructions on how to request the tracks.  See the added README.md and help text for instructions

Note: Not sure if this is the best place to put this script.  I've opened the PR for now and the script works, but we should address the destination before merging

### CODE REVIEW

**Comment by natalian98** [comment]
I think this location is good, provided that we create directories for each script (e.g. [branch track creation](https://github.com/canonical/kubeflow-ci/blob/main/scripts/branch_track_creation.py), its bundle and tests could live in its own folder rather than just `scripts`).

**Comment by ca-scribner** [comment]
Yeah agreed, I think that makes sense

File: scripts/request_missing_tracks/request_missing_tracks.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+import typer
+from bundle import Bundle
+from juju import Juju, JujuFailedError
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+def get_charm_channel_map_for_applications(applications: dict[str, dict]) -> dict[str, dict]:
```
Comment (DnPlas) [inline scripts/request_missing_tracks/request_missing_tracks.py:17]: You'd get the following error:

```
$ python3 ./request_missing_tracks.py bundle.yaml 
Traceback (most recent call last):
  File "./request_missing_tracks.py", line 17, in <module>
    def get_charm_channel_map_for_applications(applications: dict[str, dict]) -> dict[str, dict]:
TypeError: 'type' object is not subscriptable
```

I think you have to change this to something like

```suggestion
from typing import Dict
...
def get_charm_channel_map_for_applications(applications: Dict[str, dict]) -> Dict[str, dict]:
```

Tested on python3.8

**Review by DnPlas — COMMENTED** [review]
This is a really nice addition, thanks @ca-scribner. Just a tiny thing happening with typing, but other than that, LGTM.

File: scripts/request_missing_tracks/request_missing_tracks.py (hunk 1 above)
Comment (ca-scribner) [reply scripts/request_missing_tracks/request_missing_tracks.py]: yep, I bet I was running in python 3.9! (I think 3.9 is where they made `dict` work that way)

File: scripts/request_missing_tracks/request_missing_tracks.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import typer
+from bundle import Bundle
+from juju import Juju, JujuFailedError
+
+logging.basicConfig()
+logger = logging.getLogger(__name__)
+
+
+from typing import Dict
+...
```
Comment (DnPlas) [inline scripts/request_missing_tracks/request_missing_tracks.py:18]: Whoops.

**Review by DnPlas — DISMISSED** [review]
LGTM, I corrected a minor thing added when you committed my suggestion, but it should work now. Thanks @ca-scribner !

**Pull request merged**